### PR TITLE
Support Embed Internal Links

### PR DESCRIPTION
**Pull request by zephraph**
Work to support [embed files](https://help.obsidian.md/Linking+notes+and+files/Embed+files) in Obsidian. This was a piece of work, let me tell you. 

An unfortunate thing that I've learned about obsidian flavored markdown in this process:

When embedding an image you can set a size like `![[image.png|200]]`. That'll set a width. Now, you can _also_ set alt text for the image by doing `![[image.png|test]]`. You can't do both though. It's a really weird parsing edge case.

### CODE REVIEW

File: packages/astro-md/middleware.ts [hunk 1]
```diff
@@ -6,12 +6,24 @@ import { normalizeFrontmatter } from "../my-remark";
...
+  const ulidRegex = /^[0-9A-HJKMNP-TV-Z]{26}$/i;
+  return ulidRegex.test(str);
+}
 
-export const onRequest = defineMiddleware((context, next) => {
+export const onRequest = defineMiddleware(async (context, next) => {
+  const fileResolver = async (path: string) => {
+    if (isULID(path)) return path;
+    return (await context.locals.runtime.env.KV_MAPPINGS.get(path)) ?? path;
+  };
```
Comment (zephraph) [inline packages/astro-md/middleware.ts:19]: This is used to resolve asset paths.

File: packages/remark-obsidian/embed/html.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      return;
+    }
+
+    const width = safeTpl`width="${embed.dimensions?.[0]}" `;
+    const height = safeTpl`height="${embed.dimensions?.[1]}" `;
+
+    this.tag(
+      `<embed src="${embed.value}" type="${mime.getType(
+        embed.extension
+      )}" ${width}${height}/>`
```
Comment (zephraph) [inline packages/remark-obsidian/embed/html.ts:95]: These embed tags could likely get updated to `object` tags, but it doesn't _really_ matter. This isn't used.

**Comment by zephraph** [comment]
image
